app/src/UI/Features/Records/RecordSet/RecordSet.tsx

```diff
@@ -74,6 +75,10 @@ export const RecordSet = ({ setID }: PropTypes) => {
     setFilters(disabledFilters);
   }, [cacheFilters, recordSet?.tableFilters]);
 
+  useEffect(() => {
+    dispatch(Activity.switchRecordSet({ type: 'Activity', setId: parseInt(setID) }));
```

Comment:
```suggestion
    dispatch(Activity.switchRecordSet({ type: 'Activity', setId: setID }));
```

As referenced in the top level comment